server.py

```
@@ -0,0 +1,255 @@
+                        break
+
+    except websockets.exceptions.ConnectionClosed:
+        logger.info("Client connection closed.")
+    except genai.errors.GoogleAPIError as e:
+        logger.error(f"Google API Error during live session setup or operation: {e}", exc_info=True)
+        await send_to_client(websocket, {"type": "error", "data": {"message": f"AI service error: {e}"}})
+    except Exception as e:
+        logger.error(f"An unexpected error occurred in the handler: {e}", exc_info=True)
+        # Ensure client is notified if websocket is still open
+        if websocket.open:
+             await send_to_client(websocket, {"type": "error", "data": {"message": "An unexpected server error occurred."}})
```

Pylint (3.3.7)

[convention] Line too long (124/100)

(C0301)